com.unity.shadergraph/Editor/Drawing/Controllers/ShaderInputViewController.cs

```diff
@@ -244,27 +251,46 @@ protected override void ModelChanged(GraphData graphData, IGraphDataAction chang
             switch (changeAction)
             {
                 case ChangeExposedFlagAction changeExposedFlagAction:
-                    ViewModel.isInputExposed = Model.generatePropertyBlock;
-                    DirtyNodes(ModificationScope.Graph);
-                    m_SgBlackboardField.UpdateFromViewModel();
+                    // ModelChanged is called overzealously on everything
+                    // but we only care if the action pertains to our Model
+                    if (changeExposedFlagAction.shaderInputReference == Model)
+                    {
+                        ViewModel.isInputExposed = Model.generatePropertyBlock;
+                        if (changeExposedFlagAction.oldIsExposedValue != changeExposedFlagAction.newIsExposedValue)
+                            DirtyNodes(ModificationScope.Graph);
+                        m_SgBlackboardField.UpdateFromViewModel();
+                    }
                     break;
 
                 case ChangePropertyValueAction changePropertyValueAction:
-                    DirtyNodes(ModificationScope.Graph);
-                    m_SgBlackboardField.MarkDirtyRepaint();
+                    if (changePropertyValueAction.shaderInputReference == Model)
+                    {
+                        DirtyNodes(ModificationScope.Graph);
+                        m_SgBlackboardField.MarkDirtyRepaint();
+                    }
                     break;
 
                 case ResetReferenceNameAction resetReferenceNameAction:
-                    DirtyNodes(ModificationScope.Graph);
+                    if (resetReferenceNameAction.shaderInputReference == Model)
+                    {
+                        DirtyNodes(ModificationScope.Graph);
+                    }
                     break;
 
                 case ChangeReferenceNameAction changeReferenceNameAction:
-                    DirtyNodes(ModificationScope.Graph);
+                    if (changeReferenceNameAction.shaderInputReference == Model)
+                    {
+                        DirtyNodes(ModificationScope.Graph);
+                    }
                     break;
+
                 case ChangeDisplayNameAction changeDisplayNameAction:
-                    ViewModel.inputName = Model.displayName;
-                    DirtyNodes(ModificationScope.Topological);
-                    m_SgBlackboardField.UpdateFromViewModel();
+                    if (changeDisplayNameAction.shaderInputReference == Model)
+                    {
+                        ViewModel.inputName = Model.displayName;
+                        DirtyNodes(ModificationScope.Topological);
+                        m_SgBlackboardField.UpdateFromViewModel();
+                    }
```

Comment:
Can this not be done as:
`if (changeDisplayNameAction.shaderInputReference != Model)
 {
      return;
 }`
outside of the switch?  You'd still want to preserve the additional condition in the changeExposedFlagAction case (only call DirtyNodes if the value changed).

Reply:
Only if we modified the interface to require that all actions implement the shaderInputReference property... I don't think all of them operate on ShaderInputs..

Reply:
We could clean it up further to make it nice, but all of this code is going to die in GTF...

Reply:
Gotcha.  Good to go.